www/js/controller.js

```diff
@@ -692,6 +700,335 @@ define([
 			return id;
 		},
 
+		equationAnalysis: function(directives, ignoreUnknownTest){
+			this.equationEntered = true;
+			console.log("****** enter button");
+			/*
+			 This takes the contents of the equation box and parses it.
+
+			 If the parse fails:
+			 * send a warning message, and
+			 * log attempt (the PM does not handle syntax errors).
+
+			 Note: the model module may do some of these things automatically.
+
+			 Also, the following section could just as well be placed in the PM?
+			 */
+			var widget = registry.byId(this.controlMap.equation);
+			var inputEquation = widget.get("value");
+
+			var parse = {};
+			if (inputEquation == "") {
+				directives.push({id: 'message', attribute: 'append', value: 'There is no equation to check.'});
+				return null;
+			}
+			try{
+				//input equation in topomath should contain symbol "=" 
+				//both left hand side and right hand side have to be parsed
+				var splitEq = inputEquation.split("=");
+				if(splitEq.length !== 2){
+					directives.push({id: 'message', attribute: 'append', value: 'Incorrect equation syntax'});
+					return null;
+				}
+				var lhsEquation = splitEq[0];
+				var rhsEquation = splitEq[1];
+				console.log(inputEquation, typeof inputEquation, lhsEquation,typeof lhsEquation);				
```

Comment:
would this be possible to move this part of the code to equation.js. I had created a function called convert in equation.js which takes equation with id's and converts it to equation with variable names. This will make this part reusable in case we need it somewhere else.